### PR TITLE
chore: update api compliance 

### PR DESCRIPTION
**Pull request by raulvillora**
Hello!

We are migrating to a new image for doing API compliance.
Check that the paths and names are correct to your repo.

If you have any questions don't hesitate to reach out.

### CODE REVIEW

**Review by veinfors — APPROVED** [review]

File: .circleci/config.yml
Comment (cbt1) [inline .circleci/config.yml:85]: Should it not be `./package.json`?

**Comment by cbt1** [comment]
So this will run on every PR. Meaning that if you intentionally make a breaking change you build will go RED. Which seems wrong to me. You have to be allowed to make a breaking without bumping the version number, as the version bumping is done when we release a new version.

So with that in mind, would it not make more sense to have to step as part of the release script instead?

@veinfors @raulvillora

**Comment by veinfors** [comment]
> So this will run on every PR. Meaning that if you intentionally make a breaking change you build will go RED. Which seems wrong to me. You have to be allowed to make a breaking without bumping the version number, as the version bumping is done when we release a new version.
> 
> So with that in mind, would it not make more sense to have to step as part of the release script instead?
> 
> @veinfors @raulvillora

Good point! Agree that it should only be triggered when a release is to be published, that is here in the config.yaml file I guess:

```
...
if [ "$TAG_NAME" == "$PROJ_VER" ]
  then
    echo "Running >> npm publish"
    npm publish
fi
...
```

**Comment by cbt1** [comment]
Actually I was thinking at that point it's "too late". Since you have already done the release except published the package to npm. So it might make more sense to have the check running locally when you try and create a new version.

**Comment by raulvillora** [comment]
I guess it depends, some teams prefer to know if a commit is causing breaking changes so it's easier to track and fix. Otherwise, if there is a violation made in a commit and the check is only before a release is made will block the release until the violations are fixed. So as I said it depends on when the team wants to be aware of it.

Also, we will need to have this **production-approval/compliance/api-governance: required** regardless of where it's placed

**Comment by cbt1** [comment]
> I guess it depends, some teams prefer to know if a commit is causing breaking changes so it's easier to track and fix. Otherwise, if there is a violation made in a commit and the check is only before a release is made will block the release until the violations are fixed. So as I said it depends on when the team wants to be aware of it.
> 
> Also, we will need to have this **production-approval/compliance/api-governance: required** regardless of where it's placed

Right so I understand the need for it. I am more concerned about the way it affects the workflow. There are breaking changes that are intended and those that are not. In the first case an intended breaking change will cause all other PRs after to go red, which is not ideal. But on the other hand publishing an unintended breaking change is not great either..

One solution could be to do as @veinfors suggested with a check if the commit is a release commit and only then run the api gov step. But at which point is too late as the new version is already tagged and possible published.

I'll think some more about it, but it might just be that we go ahead and run this for every PR.

**Comment by raulvillora** [comment]
@cbt1  thanks for checking about this 🙇

**Review by cbt1 — APPROVED** [review]
Let go with running it on every PR. We'll have to revisit this if it becomes an issue.

File: .circleci/config.yml
Comment (raulvillora) [reply .circleci/config.yml]: You were right @cbt1 , that was making the version to be empty. Now it's fixed :)

**Pull request merged**
